crypto/fipsmodule/sha/sha3.c

```diff
@@ -236,7 +257,62 @@ int SHAKE_Init(KECCAK1600_CTX *ctx, size_t block_size) {
   return 0;
 }
 
+int SHAKE_Absorb(KECCAK1600_CTX *ctx, const void *data, size_t len) {
+  if (ctx == NULL) {
+    return 0;
+  }
+
+  if (len == 0) {
+    return 1;
+  }
+
+  return FIPS202_Update(ctx, data, len);
+}
+
+// SHAKE_Final is a single-shot API and can be called once to finalize absorb and squeeze phases
```

Comment:
```suggestion
// SHAKE_Final is to be called once to finalize absorb and squeeze phases
```
I think "single-shot" here is not the right term because it's used with non-streaming APIs such as `SHAKE_256()`